### ISSUE TITLE
mysql_user host_all arguments conversion string formatting error

### ISSUE THREAD

**Issue by meggnas** [ansible/ansible-modules-core#4894]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

mysql_user
##### ANSIBLE VERSION

```
ansible 2.1.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### OS / ENVIRONMENT

Ubuntu Trusty/Xenial
##### SUMMARY

Attempting to use the mysql_user module to change the password for the root user while using the 'host_all: yes' instead of iterating all hosts with 'host: "{{ item }}"' +  'with_items' results in error "not all arguments converted during string formatting"
##### STEPS TO REPRODUCE

```
  - name: Change password
    mysql_user:
      login_user: "root"
      login_password: "abc"
      name: "root"
      password: "xyz"
      host_all: yes
```

or

```
  - name: Change password
    mysql_user:
      name: "root"
      password: "xyz"
      host_all: yes
```

##### EXPECTED RESULTS

The mysql_user should be able to iterate through mysql hosts and change the password of the root user against hosts in mysql similar to when using the 'host: "{{ item }}"' +  'with_items'
##### ACTUAL RESULTS

```
fatal: [192.168.1.127]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_name": "mysql_user"}, "module_stderr": "OpenSSH_6.6.1, OpenSSL 1.0.1f 6 Jan 2014\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 27404\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\nShared connection to 192.168.1.127 closed.\r\n", "module_stdout": "\r\nTraceback (most recent call last):\r\n  File \"/tmp/ansible_BJCDub/ansible_module_mysql_user.py\", line 582, in <module>\r\n    main()\r\n  File \"/tmp/ansible_BJCDub/ansible_module_mysql_user.py\", line 554, in main\r\n    if user_exists(cursor, user, host, host_all):\r\n  File \"/tmp/ansible_BJCDub/ansible_module_mysql_user.py\", line 216, in user_exists\r\n    cursor.execute(\"SELECT count(*) FROM user WHERE user = %s\", user)\r\n  File \"/usr/local/lib/python2.7/dist-packages/MySQLdb/cursors.py\", line 187, in execute\r\n    query = query % tuple([db.literal(item) for item in args])\r\nTypeError: not all arguments converted during string formatting\r\n", "msg": "MODULE FAILURE", "parsed": false}
```

**Comment by ansibot** [comment]
@Jmainguy ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@Jmainguy, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by 13am** [comment]
I see the same exact error.

Control machine:
Fedora release 23 (Twenty Three)
ansible 2.1.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides

Remote node:
Ubuntu Xenial (16.04 LTS) (GNU/Linux 4.5.7-std-3 x86_64 )

**Comment by rmpratt1** [comment]
I just corrected this issue on my side in the file mysql_user.py

Line 355:

```
add [] around the user parameter
cursor.execute("SELECT Host FROM mysql.user WHERE user = %s", [user])
```

Line 216:

```
add [] around the user parameter
cursor.execute("SELECT count(*) FROM user WHERE user = %s", [user])
```

**Comment by ansibot** [comment]
@Jmainguy, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by rmpratt1** [comment]
This looks like it is an issue between python 2.6 and 2.7 it may potentially be a mac/linux issue as well. I'm not going to pile on a bunch of copies of python on my machine and a couple of machines so I'm leaving this as wontfix

**Comment by ansibot** [comment]
@Jmainguy, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@Jmainguy, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
This issue was migrated to https://github.com/ansible/ansible/issues/29644
